apps/cowswap-frontend/src/legacy/components/Header/AccountElement/index.tsx

```diff
@@ -44,18 +49,25 @@ export function AccountElement({ className, standaloneMode, pendingActivities }:
   return (
     <>
       <Wrapper className={className} active={!!account}>
-        {standaloneMode !== false && account && !isChainIdUnsupported && userEthBalance && chainId && !isUpToLarge && (
-          <BalanceText>
-            <TokenAmount amount={userEthBalance} tokenSymbol={{ symbol: nativeTokenSymbol }} />
-          </BalanceText>
-        )}
+        {!!hideNetworkSelector &&
+          standaloneMode !== false &&
+          account &&
+          !isChainIdUnsupported &&
+          userEthBalance &&
+          chainId &&
+          !isUpToLarge && (
```

Comment:
These conditions are getting out of hand.
How about a variable/fn to abstract it away?

Reply:
I cannot agree more